moment_kinetics/src/fokker_planck.jl

```diff
@@ -107,36 +108,42 @@ function init_fokker_planck_collisions_weak_form(vpa,vperp,vpa_spectral,vperp_sp
             println("finished YY array calculation   ", Dates.format(now(), dateformat"H:MM:SS"))
         end
     end
+
+    # The following velocity-space-sized buffer arrays are used to evaluate the
+    # collision operator for a single species at a single spatial point. They are
+    # shared-memory arrays. The `comm` argument to `allocate_shared_float()` is used to
+    # set up the shared-memory arrays so that they are shared only by the processes on
+    # `comm_anyv_subblock[]` rather than on the full `comm_block[]`. This means that
+    # different subblocks that are calculating the collision operator at different
+    # spatial points do not interfere with each others' buffer arrays.
     nvpa, nvperp = vpa.n, vperp.n
-    nc = nvpa*nvperp
-    S_dummy = allocate_shared_float(nvpa,nvperp)
-    Q_dummy = allocate_shared_float(nvpa,nvperp)
-    rhsvpavperp = allocate_shared_float(nvpa,nvperp)
-    rhsc = allocate_shared_float(nc)
-    rhqc = allocate_shared_float(nc)
-    sc = allocate_shared_float(nc)
-    qc = allocate_shared_float(nc)
+    S_dummy = allocate_shared_float(nvpa,nvperp; comm=comm_anyv_subblock[])

```

Comment:
I think I hadn't fully understood the documentation https://mabarnes.github.io/moment_kinetics/previews/PR176/developing/#Collision-operator-and-anyv-region, that the anyv region still has a range of iz,ir,is indices but the buffers are only nvpa*nvperp sized because we still do the spatial/species points in serial once we are on the sub block. I think I am happy with this now, so just linking to the anyv documentation as you suggest is probably adequate. We'll have to test this code on other people to get a proper idea of how easy or hard it is to follow!